install.sh

```diff
@@ -56,8 +56,12 @@ if ! sudo apt-get update; then
     die_cmd "Failed to update APT cache."
 fi
 
-if ! sudo apt-get install gpg wget; then
-    die_cmd "Failed to check if needed packages are installed."
+missing_dependencies=()
+dpkg-query -W 'wget' > /dev/null 2>&1 || missing_dependencies+=('wget')
+dpkg-query -W 'gpg' > /dev/null 2>&1 || missing_dependencies+=('gpg')
+
+if ! ( test -z "${missing_dependencies[@]}" || sudo apt-get install --mark-auto "${missing_dependencies[@]}" ); then
```

Comment:
Your `test -z` command isn't working due to an extra argument from the `[@]` stuff, you probably want this instead:

```sh
if ! ( test -z "${missing_dependencies[*]}" ...
```

Reply:
Your code wasn't working for me, no. Were you saying my example wasn't?